chore: add SECURITY.md (#2690)
tact-lang/tact: SECURITY.md

# Security policy

The Tact team and community take security bugs in Tact seriously. We appreciate your efforts to disclose your findings responsibly and will make every effort to acknowledge your contributions and fix your findings as soon as possible.

## Supported Versions

We currently support the latest stable release of the Tact compiler and infrastructure libraries.

| Versions | Supported |
| -------- | --------- |
| Latest   | ✅ Yes    |

> | 1.6.x    | ✅ Yes    |

Reviewer: Do we need to add some check in RELEASE.md, such that this table won't go out of date?

Author: good point, added

| < 1.6    | ❌ No     |

You can check the latest Tact compiler version at <https://www.npmjs.com/package/@tact-lang/compiler>.

## Reporting Security Issues

If you discover a security vulnerability in the Tact compiler, standard library, tooling, or related infrastructure, **please report it privately and responsibly**.

### How to Report

Please send us a detailed report using either email or the GitHub Security Advisory ["Report a Vulnerability"](https://github.com/tact-lang/tact/security/advisories/new) tab.

> 📧 **Email:** `it+tact@tonstudio.io@tact-lang.org`

Reviewer 2: looks strange

> 📧 **Email:** `it+tact@tonstudio.io@tact-lang.org`

Author: ```suggestion
📧 **Email:** `it+tact@tonstudio.io`
```

- the Tact language server version,
- IDE or text editor, if applicable,
- any relevant details about your execution environment, like the Node.js version or your operating system.

Please include steps to reproduce the bug you have found in as much detail as possible.

We will acknowledge your report within three business days and aim to provide a resolution or mitigation plan within 30 days.

## In scope

This policy applies to:

- The Tact compiler, including the TypeScript wrappers it generates,
- The Tact standard library (`stdlib`),
- Developer tooling: Tact's CLI, VS Code plugin, and the Tact [language server](https://github.com/tact-lang/tact-language-server),
- The Tact [documentation](https://docs.tact-lang.org),
- Web-based compiler services, such as [TON Web IDE](https://ide.ton.org),
- Smart contract generation and deployment utilities.

A security issue, in this case, may stem from miscompilation, incorrect documentation description, incorrect language server, editor plugin, or TON Web IDE suggestions, which can lead to monetary loss if an affected smart contract is deployed in the mainnet and a malicious third party interacts with it.

## Out of scope

> Only the targets listed under in-scope are part of the Security Policy and its related Bug Bounty program. This means that, for example, our infrastructure, such as webpages, is not part of the bounty scope.

Reviewer: ```suggestion
Only the targets listed under ["in scope"](#in-scope) are part of the Security Policy and its related Bug Bounty program. This means that, for example, our infrastructure, such as webpages, is not part of the bounty scope.
```

- [Reporting Security Issues](#reporting-security-issues)
  - [How to Report](#how-to-report)
- [In scope](#in-scope)
- [Out of scope](#out-of-scope)
- [Responsible Disclosure](#responsible-disclosure)
- [PGP key to use if reporting via email](#pgp-key-to-use-if-reporting-via-email)

## Supported Versions

We currently support the latest stable release of the Tact compiler and infrastructure libraries.

| Versions | Supported |
| -------- | --------- |
| Latest   | ✅ Yes    |
| 1.6.x    | ✅ Yes    |
| < 1.6    | ❌ No     |

You can check the latest Tact compiler version at <https://www.npmjs.com/package/@tact-lang/compiler>.

## Reporting Security Issues

If you discover a security vulnerability in the Tact compiler, standard library, tooling, or related infrastructure, **please report it privately and responsibly**.

### How to Report

Please send us a detailed report using either email or the GitHub Security Advisory ["Report a Vulnerability"](https://github.com/tact-lang/tact/security/advisories/new) tab.

📧 **Email:** `it+tact@tonstudio.io`
🔒 Please encrypt your email using our [PGP key](#pgp-key-to-use-if-reporting-via-email).

Your report should include:

> - A description of the vulnerability.

Reviewer 3: nit: Let's make formatting of the lists more consistent (e.g. capitalized, no trailing comma)

- Steps to reproduce (if applicable).
- The potential impact.
- Any suggestions for remediation (if known).

Please include all details necessary to reproduce the vulnerability, such as:

- the input program that triggers the bug,
- compiler version affected,
- target TVM version,
- the link to the mainnet or testnet transaction that demonstrates the bug,
- if reproducing the bug using local emulators, such as [Sandbox](https://github.com/ton-org/sandbox), please add the test script, the instructions to install its dependencies, and the command to run the script to reproduce the issue,
- the Tact language server version,
- IDE or text editor, if applicable,
- any relevant details about your execution environment, like the Node.js version or your operating system.

Please include steps to reproduce the bug you have found in as much detail as possible.

We will acknowledge your report within three business days and aim to provide a resolution or mitigation plan within 30 days. If a resolution requires more time, we will keep you updated on the progress.

## In scope

> This policy applies to:

Reviewer 3: How about CI? And npm repositories? These are very common attack vectors.

- The Tact compiler, including the TypeScript wrappers it generates,
- The Tact standard library (`stdlib`),

> - Developer tooling: Tact's CLI, VS Code plugin, and the Tact [language server](https://github.com/tact-lang/tact-language-server),

Reviewer 3: I would create a complete list with links to github repos for clarity.

Reviewer 3: I'm not sure if I got it. CLI or CI? No mistake here?

- The Tact [documentation](https://docs.tact-lang.org),
- Web-based compiler services, such as [TON Web IDE](https://ide.ton.org),
- Smart contract generation and deployment utilities.

> A security issue, in this case, may stem from miscompilation, incorrect documentation description, incorrect language server, editor plugin, or TON Web IDE suggestions, which can lead to monetary loss if an affected smart contract is deployed in the mainnet and a malicious third party interacts with it.

Reviewer 3: I would replace this with severity level descriptions. Attacks leading to DoS attacks, gas draining or similar issues are problematic but definitely not critical.

Reviewer 3: Also, would be nice to have a bullet-list here as well.

## Out of scope

> Only the targets listed under in-scope are part of the security policy. This means that, for example, our infrastructure, such as webpages, is not part of the scope.

Reviewer 3: I would explicitly specify what is out of scope:

```suggestion
Only the targets listed under "in-scope" are part of the security policy.

The following are considered out of scope:  
- Server infrastructure (TODO: backend components, static webpages, ...)  
- ...
```
